pkg/daemon/ceph/osd/volume.go

```diff
@@ -118,14 +119,22 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 				logger.Infof("using %s as metadataDevice for device %s and let ceph-volume lvm batch decide how to create volumes", md, deviceArg)
 				if _, ok := metadataDevices[md]; ok {
 					metadataDevices[md] = append(metadataDevices[md], deviceArg)
+					// Fail when two devices using the same metadata device have different values for osdsPerDevice
+					if osdsPerDeviceCount != metadataDevices[md][1] {
```

Comment:
Should you be comparing against index 0? Seems like index 1 will be the item you added the second time. And will there always be two entries? Otherwise this could crash if there is only 1.

Reply:
Each metadataDevice key in `metadataDevices` is being initialized as:
```
213                                         metadataDevices[md] = []string{
214                                                 osdsPerDeviceFlag,
215                                                 deviceOSDCount,
216                                                 deviceArg,
217                                         }
```
So 0 will always be the `osdsPerDeviceFlag` and 1 will be the `deviceOSDCount`